It looks like we do have the exactly same issue as FreeBSD
https://github.com/python/cpython/issues/133558

--- Python-3.14.2/Lib/test/test_pdb.py.orig
+++ Python-3.14.2/Lib/test/test_pdb.py
@@ -4924,6 +4924,8 @@
 
     @unittest.skipIf(sys.platform.startswith('freebsd'),
                      '\\x08 is not interpreted as backspace on FreeBSD')
+    @unittest.skipIf(sys.platform.startswith('sunos'),
+                     '\\x08 is not interpreted as backspace on SunOS')
     def test_multiline_auto_indent(self):
         script = textwrap.dedent("""
             import pdb; pdb.Pdb().set_trace()
@@ -4964,6 +4966,8 @@
 
     @unittest.skipIf(sys.platform.startswith('freebsd'),
                      '\\x08 is not interpreted as backspace on FreeBSD')
+    @unittest.skipIf(sys.platform.startswith('sunos'),
+                     '\\x08 is not interpreted as backspace on SunOS')
     def test_multiline_indent_completion(self):
         script = textwrap.dedent("""
             import pdb; pdb.Pdb().set_trace()
